### PR TITLE
feat(Shared): Migrated Remaining Bridge Files

### PR DESCRIPTION
**Pull request by kevinvangelder**

### CODE REVIEW

File: ReactWindows/ReactNative.Shared/Bridge/JavaScriptBundleLoader.cs [hunk 1]
```diff
@@ -1,6 +1,10 @@
 ﻿using System;
 using System.Threading.Tasks;
+#if WINDOWS_UWP
 using Windows.Storage;
+#else
+using PCLStorage;
+#endif
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/Bridge/JavaScriptBundleLoader.cs]: Don't we already have a PCLStorage dependency on UWP?

Comment (rozele) [inline ReactWindows/ReactNative.Shared/Bridge/JavaScriptBundleLoader.cs]: If we have the dependency anyway, we should just use it here.

---
In reply to: [85593448](https://github.com/ReactWindows/react-native-windows/pull/835#discussion_r85593448) [](ancestors = 85593448)

File: ReactWindows/ReactNative.Shared/Bridge/CachedJavaScriptBundleLoader.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading.Tasks;
+using PCLStorage;
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/Bridge/CachedJavaScriptBundleLoader.cs]: Sort.

File: ReactWindows/ReactNative.Shared/Bridge/CachedJavaScriptBundleLoader.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            {
+                var localFolder = FileSystem.Current.LocalStorage;
+                var storageFile = await localFolder.GetFileAsync(_cachedFileLocation).ConfigureAwait(false);
+                _script = storageFile.Path;
+            }
+            catch (Exception ex)
+            {
+                var exceptionMessage = FormattableString.Invariant($"File read exception for asset '{SourceUrl}'.");
+                throw new InvalidOperationException(exceptionMessage, ex);
+            }
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/Bridge/CachedJavaScriptBundleLoader.cs]: I wonder if it's even worth catching and throwing a different exception here, or if we should just let the original exception go.

File: ReactWindows/ReactNative.Shared/Bridge/FileJavaScriptBundleLoader.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+#else
+                var assembly = Assembly.GetAssembly(typeof(FileJavaScriptBundleLoader));
+                var assemblyName = assembly.GetName();
+                var pathToAssembly = Path.GetDirectoryName(assemblyName.CodeBase);
+                var pathToAssemblyResource = Path.Combine(pathToAssembly, SourceUrl.Replace("ms-appx:///", String.Empty));
+                var u = new Uri(pathToAssemblyResource);
+                _script = u.LocalPath;
+#endif
+            }
+            catch (Exception ex)
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/Bridge/FileJavaScriptBundleLoader.cs]: >Exception [](start = 19, length = 9)

Another case of "do we really need to do this in a try/catch or is the original exception okay"

File: ReactWindows/ReactNative.Shared/Bridge/FileJavaScriptBundleLoader.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            try
+            {
+#if WINDOWS_UWP
+                var storageFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri(SourceUrl)).AsTask().ConfigureAwait(false);
+                _script = storageFile.Path;
+#else
+                var assembly = Assembly.GetAssembly(typeof(FileJavaScriptBundleLoader));
+                var assemblyName = assembly.GetName();
+                var pathToAssembly = Path.GetDirectoryName(assemblyName.CodeBase);
+                var pathToAssemblyResource = Path.Combine(pathToAssembly, SourceUrl.Replace("ms-appx:///", String.Empty));
```
Comment (rozele) [inline ReactWindows/ReactNative.Shared/Bridge/FileJavaScriptBundleLoader.cs]: >(" [](start = 91, length = 2)

I actually don't really like this approach.  Why would you use "ms-appx:///" as a protocol for Windows 7?  Each platform gets to define it's own URI protocol.  Can't you just use the [file:///](file:///) for WPF?

**Comment by rozele** [comment]
>         class FileJavaScriptBundleLoader : JavaScriptBundleLoader

Any particular reason for splitting these out?

---
Refers to: ReactWindows/ReactNative.Shared/Bridge/JavaScriptBundleLoader.cs:75 in 46b73b3. [](commit_id = 46b73b3acaa9c58fd53f99636ce16450b1786440, deletion_comment = True)

**Comment by rozele** [comment]
>         class FileJavaScriptBundleLoader : JavaScriptBundleLoader

Everything is in Shared already anyway.

---
In reply to: [257302219](https://github.com/ReactWindows/react-native-windows/pull/835#issuecomment-257302219) [](ancestors = 257302219)

---
Refers to: ReactWindows/ReactNative.Shared/Bridge/JavaScriptBundleLoader.cs:75 in 46b73b3. [](commit_id = 46b73b3acaa9c58fd53f99636ce16450b1786440, deletion_comment = True)

File: ReactWindows/ReactNative.Shared/Bridge/CachedJavaScriptBundleLoader.cs (hunk 3 above)
Comment (matthargett) [inline ReactWindows/ReactNative.Shared/Bridge/CachedJavaScriptBundleLoader.cs]: Since that was in the original code, can we handle it in a separate PR? At that point, we can also write a unit test that pins the exception message behavior.

File: ReactWindows/ReactNative.Shared/Bridge/FileJavaScriptBundleLoader.cs (hunk 4 above)
Comment (matthargett) [inline ReactWindows/ReactNative.Shared/Bridge/FileJavaScriptBundleLoader.cs]: Same  reply as above -- it already existed, so can we do work and add test in a separate PR?

File: ReactWindows/ReactNative.Shared/Bridge/FileJavaScriptBundleLoader.cs (hunk 5 above)
Comment (matthargett) [inline ReactWindows/ReactNative.Shared/Bridge/FileJavaScriptBundleLoader.cs]: ms-appx URLs are hard-coded in several of the tests and we were trying to be less invasive since it' was easy to translate transparently and have the tests stay passing. We can absolutely make a test-helper that abstracts this, or take another approach.

**Comment by matthargett** [comment]
I originally split them in a different PR because:
1. There was a #if and possible future divergence, and didn't want to drag the rest of the classes along for that ride 
2. The static creation methods don't actually encapsulate because there isn't an abstract factory to truly break the coupling
3. The creation methods were straight delegation of arguments (no behavior), and only used in one place. Without the benefit of real encapsulation/decoupling, it ends up being indirection with no value.
4. We got feedback on a PR earlier to not put multiple classes in a file if we can help it, to be sensitive to Java PTSD ;)

**Comment by rozele** [comment]
Multiple public classes*  nested classes are private by default, I see these as implementation details and not needing to be split out.

---
In reply to: [257308345](https://github.com/ReactWindows/react-native-windows/pull/835#issuecomment-257308345) [](ancestors = 257308345)

File: ReactWindows/ReactNative.Shared/Bridge/FileJavaScriptBundleLoader.cs (hunk 4 above)
Comment (rozele) [inline ReactWindows/ReactNative.Shared/Bridge/FileJavaScriptBundleLoader.cs]: Yes.

---
In reply to: [85747268](https://github.com/ReactWindows/react-native-windows/pull/835#discussion_r85747268) [](ancestors = 85747268)

File: ReactWindows/ReactNative/Bridge/JavaScriptBundleLoader.cs [hunk 6]
```diff
@@ -1,179 +0,0 @@
-﻿using System;
-using System.Threading.Tasks;
-using Windows.Storage;
-using static System.FormattableString;
-
-namespace ReactNative.Bridge
-{
```
Comment (rozele) [inline ReactWindows/ReactNative/Bridge/JavaScriptBundleLoader.cs]: Can you double check if squashing these commits makes this a file rename instead of a delete and add?

Comment (kevinvangelder) [inline ReactWindows/ReactNative/Bridge/JavaScriptBundleLoader.cs]: It was originally a file move but GitHub shows it as a delete/add if you change any lines.
